controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -229,23 +229,31 @@ func (p *OpenAPIProductReconciler) desiredDeployment() *capabilitiesv1beta1.Prod
 }
 
 func (p *OpenAPIProductReconciler) desiredAuthentication() *capabilitiesv1beta1.AuthenticationSpec {
-	globalSecRequirements := helper.OpenAPIGlobalSecurityRequirements(p.openapiObj)
-	if len(globalSecRequirements) == 0 {
-		// if no security requirements are found, default to UserKey auth
-		return p.desiredUserKeyAuthentication(nil)
-	}
+	var authenticationSpec *capabilitiesv1beta1.AuthenticationSpec
 
-	// Only the first one is used
-	secRequirementExtended := globalSecRequirements[0]
+	if p.openapiCR.Spec.SecurityScheme != nil &&
```

Comment:
the security scheme is defined in the OpenAPI document, not  in the OpenAPI CR. If the OAS doc does not have global security requirement, 3scale product should be "open". Check out how 3scale product is configured when there is no global sec requirement.

Reply:
Thank you very much for these notes, yes, I was confused here.  Changed